sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidOuterQueryRel.java

```diff
@@ -116,12 +114,20 @@ public DruidQuery toDruidQuery(final boolean finalizeAggregations)
   @Override
   public DruidQuery toDruidQueryForExplaining()
   {
+    final RowSignature sourceRowSignature;
+    if (sourceRel instanceof DruidRel) {
+      final DruidQuery subQuery = ((DruidRel) sourceRel).toDruidQueryForExplaining();
+      sourceRowSignature = subQuery.getOutputRowSignature();
+    } else {
+      // fallback for cases other than group by inner query with order by with alias on __time column
+      sourceRowSignature = RowSignatures.fromRelDataType(
+          sourceRel.getRowType().getFieldNames(),
+          sourceRel.getRowType()
```

Comment:
Why does the `sourceRel` return the wrong row type? A naive reading of this says that we ask the source rel for its row type. Each rel can have only one "output" row type. Why do we get the wrong one if the source rel is a `DruidRel`? Is there a bug earlier in the food chain? Said another way, is the `getRowType()` method wrong for `DruidRel`?

Reply:
This was a case encountered during aliasing. If a query was specified with an alias say for example `select __time as t, dim1 as d from foo` for the inner query the field name was changing to `t` from `__time` for that reason the fallback was used

Reply:
Added the test